uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/Entitlement.php

```diff
@@ -79,4 +83,89 @@ private static function validateComment($params) {
     }
   }
 
+  /**
+   * Returns the calculated entitlement for the given JobContract,
+   * AbsencePeriod and AbsenceType
+   *
+   * @param \CRM_Hrjobcontract_BAO_HRJobContract $contract
+   * @param \CRM_HRLeaveAndAbsences_BAO_AbsencePeriod $period
+   * @param \CRM_HRLeaveAndAbsences_BAO_AbsenceType $type
+   *
+   * @return \CRM_HRLeaveAndAbsences_BAO_Entitlement|null If there's no entitlement for the given arguments, null will be returned
+   */
+  public static function getContractEntitlementForPeriod(
```

Comment:
Since you are only concerned  with ( job contract , absence period , absence type ) IDs .. Why not passing the IDs of these only instead of passing the whole object for each one as a parameter .